onnxscript/rewriter/matmul_add_to_gemm_test.py

```diff
@@ -0,0 +1,314 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+import unittest
+from typing import Sequence
+
+import numpy as np
+import onnx
+from onnx_ir.passes.common import onnx_checker, shape_inference
+from parameterized import parameterized
+
+from onnxscript import ir
+from onnxscript.rewriter import matmul_add_to_gemm, testing
+from onnxscript.rewriter import pattern as orp
+from onnxscript.rewriter.matmul_add_to_gemm import matmul_add_to_gemm_rule
+
+
+class _MatMulAddToGemmTestBase(unittest.TestCase):
+    @property
+    def rng(self):
+        return np.random.default_rng(20250607)
+
+    def clone_model(self, model: ir.Model) -> ir.Model:
+        return ir.from_proto(ir.to_proto(model))
+
+    def get_test_model(
+        self,
+        input_shape: ir.Shape,
+        weight_shape: ir.Shape,
+        transA: bool = False,
+        transB: bool = False,
+        permA: Sequence[int] = [1, 0],
+        permB: Sequence[int] = [1, 0],
+        weight_as_inputs: bool = False,
+        bias_as_inputs: bool = False,
+    ):
+        """Returns the following model:
+
+            Y = Add(MatMul(Transpose(X), Transpose(W)), B)
+
+        Where:
+        - Transpose(X) is applied only if `transA=True`
+        - Transpose(W) is applied only if `transB=True`
+        - W and B can be graph inputs or initializers
+        """
+        tape = ir.tape.Tape()
+        inputs = []
+        bias_shape = weight_shape[0] if transB else weight_shape[-1]
+        output_shape = ir.Shape(("?",) * input_shape.rank())
+
+        x = ir.Input("X", shape=input_shape, type=ir.TensorType(ir.DataType.FLOAT))
+
+        if weight_as_inputs:
+            w = ir.Input("W", shape=weight_shape, type=ir.TensorType(ir.DataType.FLOAT))
+            inputs.append(w)
+        else:
+            w = ir.tensor(
+                self.rng.uniform(-0.5, 0.5, weight_shape).astype("float32"), name="W"
+            )
+            w = tape.initializer(w)
+
+        if bias_as_inputs:
+            b = ir.Input(
+                "B", shape=ir.Shape([bias_shape]), type=ir.TensorType(ir.DataType.FLOAT)
+            )
+            inputs.append(b)
+        else:
+            b = ir.tensor(self.rng.uniform(-0.5, 0.5, bias_shape).astype("float32"), name="B")
+            b = tape.initializer(b)
+
+        if transA:
+            x_t = tape.op("Transpose", inputs=[x], attributes={"perm": permA})
+
+        if transB:
+            w_t = tape.op("Transpose", inputs=[w], attributes={"perm": permB})
```

Comment:
```suggestion
        if transA:
            x = tape.op("Transpose", inputs=[x], attributes={"perm": permA})

        if transB:
            w = tape.op("Transpose", inputs=[w], attributes={"perm": permB})
```

maybe just do this to make the code checkers happy